sklearn/datasets/_base.py

```diff
@@ -264,22 +269,47 @@ def load_data(module_path, data_file_name):
         A 1D array containing the names of the classifications. For example
         target_names[0] is the name of the target[0] class.
     """
-    with open(join(module_path, "data", data_file_name)) as csv_file:
-        data_file = csv.reader(csv_file)
-        temp = next(data_file)
-        n_samples = int(temp[0])
-        n_features = int(temp[1])
-        target_names = np.array(temp[2:])
-        data = np.empty((n_samples, n_features))
-        target = np.empty((n_samples,), dtype=int)
-
-        for i, ir in enumerate(data_file):
-            data[i] = np.asarray(ir[:-1], dtype=np.float64)
-            target[i] = np.asarray(ir[-1], dtype=int)
+    with importlib_resources.path(data_module, data_file_name) as data_path:
+        with open(data_path) as csv_file:
+            data_file = csv.reader(csv_file)
+            temp = next(data_file)
+            n_samples = int(temp[0])
+            n_features = int(temp[1])
+            target_names = np.array(temp[2:])
+            data = np.empty((n_samples, n_features))
+            target = np.empty((n_samples,), dtype=int)
+
+            for i, ir in enumerate(data_file):
+                data[i] = np.asarray(ir[:-1], dtype=np.float64)
+                target[i] = np.asarray(ir[-1], dtype=int)
 
     return data, target, target_names
 
 
+def load_descr(descr_module, descr_file_name):
+    """Loads descr_file_name from descr_module with importlib.resources.path.
+
+    Parameters
+    ----------
+    descr_module : string or module
+        module where descr_file_name lives;
+
+    descr_file_name : string
+        Name of rst file to be loaded from
+        descr_module/descr_file_name. For example 'wine_data.rst'.
+
+    Returns
+    -------
+    string; content of descr_file_name
+    """
+
+    with importlib_resources.path(descr_module, descr_file_name) as data_path:
+        with open(data_path) as rst_file:
+            fdescr = rst_file.read()
```

Comment:
This can probably be rewritten as:

```suggestion
    fdescr = importlib_resources.read_text(descr_module, descr_file_name)
```

to avoid assuming at any point that `descr_file_name` is the name of an actual file on a file system.